src/runner_manager.py

```diff
@@ -522,15 +523,54 @@ def reconcile(self, quantity: int, resources: VirtualMachineResources) -> int:
             )
         return delta
 
-    def flush(self, flush_busy: bool = True) -> int:
+    def _runners_in_pre_job(self) -> bool:
+        """Check there exist runners in the pre-job script stage.
+
+        If a runner has taken a job for 1 minute or more, it is assumed to exit the pre-job script.
+
+        Returns:
+            Whether there are runners that has taken a job and run for less than 1 minute.
+        """
+        now = datetime.now(timezone.utc)
+        busy_runners = [
+            runner for runner in self._get_runners() if runner.status.exist and runner.status.busy
+        ]
+        for runner in busy_runners:
+            exit_code, stdout, _ = runner.instance.execute(
+                ["/usr/bin/stat", "-c", "'%w'", "/home/ubuntu/github-runner/_work"]
+            )
+            if exit_code != 0:
+                return False
+            date_str, time_str, timezone_str = stdout.read().decode("utf-8").split(" ")
+            job_start_time = datetime.fromisoformat(f"{date_str}T{time_str[:12]}{timezone_str}")
+            if job_start_time + timedelta(minutes=1) > now:
+                return False
+        return True
+
+    def flush(self, flush_busy: bool = True, wait_repo_check: bool = False) -> int:
         """Remove existing runners.
 
         Args:
-            flush_busy: Whether to flush busy runners as well.
+            flush_busy: Whether to flush busy runners.
+            wait_repo_check: Whether to wait for busy runner to complete
+                repo-policy-compliance check before flushing the runners.
 
         Returns:
             Number of runners removed.
         """
+        if wait_repo_check:
```

Comment:
Should the non-busy runners be flushed first so that no new runners are spawned whilst we wait for existing runners to pass their repo-policy-check phase?

Reply:
Updated to remove the idle runners first.